pkg/helpers/helpers.go

```diff
@@ -269,29 +271,79 @@ func RemoveManagedClusterFinalizer(ctx context.Context, runtimeClient client.Cli
 	return nil
 }
 
-// UpdateManagedClusterStatus update managed cluster status
-func UpdateManagedClusterStatus(client client.Client, managedClusterName string, cond metav1.Condition) error {
+// updateManagedClusterStatus update managed cluster status
+// return true if the status is updated
+func updateManagedClusterStatus(client client.Client, managedClusterName string, cond metav1.Condition) (bool, error) {
 	managedCluster := &clusterv1.ManagedCluster{}
 	err := client.Get(context.TODO(), types.NamespacedName{Name: managedClusterName}, managedCluster)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	oldStatus := &managedCluster.Status
 	newStatus := oldStatus.DeepCopy()
 
 	meta.SetStatusCondition(&newStatus.Conditions, cond)
 	if equality.Semantic.DeepEqual(managedCluster.Status.Conditions, newStatus.Conditions) {
-		return nil
+		return false, nil
 	}
 
 	managedCluster.Status = *newStatus
 	if err := client.Status().Update(context.TODO(), managedCluster); err != nil {
 		klog.Errorf("Update the managed cluster %s condition %v failed, error: %v", managedClusterName, cond, err)
+		return true, err
+	}
+
+	klog.Infof("Update the managed cluster %s condition %v succeeded.", managedClusterName, cond)
+
+	return true, nil
+}
+
+// UpdateManagedClusterImportCondition update managed cluster status and record the event
+func UpdateManagedClusterImportCondition(client client.Client, managedCluster *clusterv1.ManagedCluster,
+	cond metav1.Condition, recorder kevents.EventRecorder) error {
+	if cond.Type != constants.ConditionManagedClusterImportSucceeded {
+		return fmt.Errorf("the condition type %s is not supported", cond.Type)
+	}
+
+	changed, err := updateManagedClusterStatus(client, managedCluster.Name, cond)
+	if err != nil {
 		return err
 	}
+	if !changed {
+		return nil
+	}
 
-	klog.V(4).Infof("Update the managed cluster %s condition %v succeeded.", managedClusterName, cond)
+	mc := managedCluster.DeepCopy()
+	mc.SetNamespace(mc.Name)
+	switch cond.Reason {
+	case constants.ConditionReasonManagedClusterWaitForImporting:
+		recorder.Eventf(mc, nil, corev1.EventTypeNormal,
+			constants.EventReasonManagedClusterWait, constants.EventReasonManagedClusterWait,
+			"The %s is waiting for importing", mc.Name)
+	case constants.ConditionReasonManagedClusterImporting:
+		recorder.Eventf(mc, nil, corev1.EventTypeNormal,
+			constants.EventReasonManagedClusterImporting, constants.EventReasonManagedClusterImporting,
+			"The %s is being imported now: %s", mc.Name, cond.Message)
+	case constants.ConditionReasonManagedClusterImported:
+		recorder.Eventf(mc, nil, corev1.EventTypeNormal,
+			constants.EventReasonManagedClusterImported, constants.EventReasonManagedClusterImported,
+			"The %s is imported successfully", mc.Name)
+	case constants.ConditionReasonManagedClusterImportFailed:
```

Comment:
How many ImportFailed events will be emitted if the auto-import keeps failed.

Reply:
it will be multiple events since the retry mechanism causes the condition message changes, like `Try to import managed cluster, retry times: 1/3, error: xxx`